doc/manual/src/notifications.md

```diff
@@ -0,0 +1,42 @@
+# `hydra-notify` and Hydra's Notifications
+
+Hydra uses a notification-based subsystem to implement some features and support plugin development. Notifications are sent to `hydra-notify`, which is responsible for dispatching each notification to each plugin.
+
+Notifications are passed from `hydra-queue-runner` to `hydra-notify` through Postgres's `NOTIFY` and `LISTEN` feature.
+
+## Notification Types
+
+Note that the notification format is subject to change and should not be considered an API. Integrate with `hydra-notify` instead of listening directly.
+
+### `build_started`
+
+* **Payload:** Exactly one value, the ID of the build.
+* **When:** Issued directly before building happens, and only if the derivation's outputs cannot be subsituted.
+* **Delivery Semantics:** Ephemeral. `hydra-notify` must be running to react to this event. No record of this event is stored.
+
+### `step_finished`
+
+* **Payload:** Three values, tab separated: The ID of the build which the step is part of, the step number, and the path on disk to the log file.
+* **When:** Issued directly after a step completes, regardless of success. Is not issued if the step's derivation's outputs can be substituted.
+* **Delivery Semantics:** Ephemeral. `hydra-notify` must be running to react to this event. No record of this event is stored.
+
+### `build_finished`
+
+* **Payload:** At least one value, tab separated: The ID of the build which finished, followed by IDs of all of the builds which also depended upon this build.
+* **When:** Issued directly after a build completes, regardless of success and substitutability.
+* **Delivery Semantics:** At least once.
```

Comment:
But it emits twice if e.g. a transient failure marks a build as failed and you restart it manually, correct? (I think it may make sense to clarify this here).